src/main/java/org/emela/chautari/service/impl/ReviewServiceImpl.java

```diff
@@ -0,0 +1,141 @@
+                             final RentalService rentalService,
+                             final UserService userService) {
+        this.reviewEntityRepository = reviewEntityRepository;
+        this.reviewEntityMapper = reviewEntityMapper;
+        this.rentalService = rentalService;
+        this.userService = userService;
+
+    }
+
+    @Override
+    public ReviewResponse addReview(String rentalId, Review review) {
+
+        String userId = review.getReviewer();
+        log.info("Add review request is received for rentalId {} by userId {}", rentalId, userId);
+
+        ReviewEntity savedReviewEntity;
+        try {
+            ReviewEntity reviewEntity = reviewEntityMapper.mapToReviewEntity(review);
+            reviewEntity.setRental(rentalService.getRentalEntity(rentalId));
+            reviewEntity.setUser(userService.getUserEntity(userId));
+            savedReviewEntity = reviewEntityRepository.save(reviewEntity);
+        } catch (Exception exception) {
+            log.error("Add review operation failed for rental id {} by user id {}", rentalId, userId);
+            throw new ReviewServiceException("Add review operation failed for rental id " + rentalId + " by user id " + userId, exception);
+        }
+        return new ReviewResponse().reviewId(savedReviewEntity.getReviewId().toString());
+    }
+
+    @Override
+    public Reviews getReviews(String rentalId) {
+        log.info("Get review details request is received for rentalId {}", rentalId);
+        List<Review> reviewList = new ArrayList<>();
+        try {
+            List<ReviewEntity> reviewEntities =
+                    reviewEntityRepository.findByRental(rentalService.getRentalEntity(rentalId))
+                            .orElseThrow(ResourceNotFoundException::new);
+
+            reviewEntities.forEach(reviewEntity -> {
+                Review review = reviewEntityMapper.mapToReview(reviewEntity);
+                review.setReviewer(reviewEntity.getUser().getUserId().toString());
+                reviewList.add(review);
+            });
+            Set<String> a = new HashSet<>();
+
+        } catch (Exception exception) {
+            log.error("Get reviews failed for rentalId {}", rentalId);
+            throw new ReviewServiceException("Get reviews failed for rentalId {}" + rentalId, exception);
+        }
+        return new Reviews().reviews(reviewList).averageLike(getAverageLike(reviewList));
+    }
+
+    private BigDecimal getAverageLike(List<Review> reviewList) {
+        int totalLike =
+                reviewList.stream().map(review -> review.getStar().toString()).mapToInt(Integer::valueOf).sum();
+        return new BigDecimal(totalLike / reviewList.size()).setScale(1, RoundingMode.CEILING);
+    }
+
+    @Override
+    public Void deleteReview(String rentalId, String reviewId) {
+        log.info("Delete review request is received for rentalId {}  reviewId {}", rentalId, reviewId);
```

Comment:
we need to discuss regarding this delete review operation. User cant delete review by other.